packages/in_app_purchase/in_app_purchase_storekit/darwin/in_app_purchase_storekit/Sources/in_app_purchase_storekit/StoreKit2/InAppPurchasePlugin+StoreKit2.swift

```diff
@@ -54,7 +54,52 @@ extension InAppPurchasePlugin: InAppPurchase2API {
           return completion(.failure(error))
         }
 
-        let result = try await product.purchase(options: [])
+        var purchaseOptions: Set<Product.PurchaseOption> = []
+
+        if let appAccountToken = options?.appAccountToken,
+          let accountTokenUUID = UUID(uuidString: appAccountToken)
+        {
+          purchaseOptions.insert(.appAccountToken(accountTokenUUID))
+        }
+
+        #if compiler(>=6.0)
```

Comment:
> Maybe an #if compiler directive could help here, but I haven’t tested it yet, so I’m not sure if it actually works. Let me know what you think

We should not land code that does not compile in our CI. For example, we have no way of knowing whether these tests will actually pass in our CI, meaning they could block future Xcode updates, and we also aren't actually getting any test coverage that would prevent landing a future PR that breaks this new code or the associated tests.

If the feature requires an SDK version we don't yet have in CI, we need to wait for that SDK to be in CI to proceed.